### PR TITLE
contains hotfixes for plotting utilities

### PR DESCRIPTION
**Pull request by wsavran**
@pabloitu do you mind taking a look at this pr when you have a chance?

contains a few adjustments to the plotting utilites
* draw region border correctly
* properly scale axes for comparison test plots
* allow chaining for consistency test plots

### CODE REVIEW

File: csep/utils/plots.py [hunk 1]
```diff
@@ -1339,18 +1339,17 @@ def plot_comparison_test(results_t, results_w=None, axes=None, plot_args=None):
     ax.set_ylabel(ylabel)
     ax.set_title(title)
     ax.yaxis.grid()
-    xTickPos, _ = pyplot.xticks()
+    xTickPos = ax.get_xticks()
     ax.yaxis.set_major_locator(matplotlib.ticker.MaxNLocator(integer=True))
-    ax.set_ylim([ylims[0], ylims[1]])
-    ax.set_xlim([ax.get_xlim()[0] + 0.5, ax.get_xlim()[1] - 0.5])
+    ax.set_ylim([ylim[0], ylim[1]])
+    ax.set_xlim([ax.get_xlim()[0] - 0.5, ax.get_xlim()[1] + 0.5])
```
Comment (pabloitu) [inline csep/utils/plots.py:1345]: This ax.set_xlim() change gives an extra edge border that I'm not sure is needed. This gives
image

versus this, which was in the previous xlim definition
image

But this failed for low number of Models in the t-test. I'd suggest to leave it better like:

`ax.set_xlim([-0.5, len(results_t) - 0.5])`

It appears to work well with any number of forecasts

**Review by pabloitu — CHANGES_REQUESTED** [review]
Thanks for the check of the functions! I believe only  `plot_comparison_test() : ax.set_xlim(-0.5, len(results_t) - 0.5)`  is needed for mergning

File: csep/core/regions.py [hunk 2]
```diff
@@ -744,8 +744,8 @@ def tight_bbox(self):
             asc.insert(0, poly_min[0])
             asc.insert(0, poly_min[1])
             poly_max = self.polygons[int(row[argmax])].points
-            desc.append(poly_max[3])
             desc.append(poly_max[2])
+            desc.append(poly_max[3])
```
[the remark this answers is not in the document]
Comment (wsavran) [reply csep/core/regions.py:748]: i think this issue is from the way i am pulling out the outermost region. thanks for spotting this, i'll come up with a solution so it works for both cases.

File: csep/utils/plots.py (hunk 1 above)
Comment (wsavran) [reply csep/utils/plots.py:1345]: that doesn't seem to work with only two models, it only displays one
image
if i use 
```
ax.set_xlim([-0.5, len(results_t) + 0.5]) 
```
it seems to work ok. am i missing something here?
image

Comment (pabloitu) [reply csep/utils/plots.py:1345]: checking....

File: csep/core/regions.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (wsavran) [reply csep/core/regions.py]: i pushed a hotfix for this issue. i checked it against the built in region. everything looks to work properly. if you have this pr cloned do you mind checking as well?

File: csep/utils/plots.py (hunk 1 above)
Comment (wsavran) [reply csep/utils/plots.py:1345]: i really like the way these plots look!

Comment (pabloitu) [reply csep/utils/plots.py:1345]: Weird!  Using `ax.set_xlim([-0.5, len(results_t) - 0.5])` I got this plot correctly:
image

Comment (wsavran) [reply csep/utils/plots.py]: i think i got it. i just pushed a new commit with that fix.

File: csep/core/regions.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (pabloitu) [reply csep/core/regions.py]: Checked!  Looks perfect

File: csep/utils/plots.py [hunk 3]
```diff
@@ -1342,7 +1342,7 @@ def plot_comparison_test(results_t, results_w=None, axes=None, plot_args=None):
     xTickPos = ax.get_xticks()
     ax.yaxis.set_major_locator(matplotlib.ticker.MaxNLocator(integer=True))
     ax.set_ylim([ylim[0], ylim[1]])
-    ax.set_xlim([ax.get_xlim()[0] - 0.5, ax.get_xlim()[1] + 0.5])
+    ax.set_xlim([0.5, len(results_t) - 0.5])
```
Comment (pabloitu) [inline csep/utils/plots.py:1345]: nope, this still hides me a forecast. It works for me with [ - 0.5, len(results_t) - 0.5].  Weiirdd.

**Review by pabloitu — APPROVED** [review]
Will be back tomorrow early. GL

File: csep/utils/plots.py (hunk 3 above)
Comment (wsavran) [reply csep/utils/plots.py:1345]: missed the '-'. catch you tomorrow! gn!

**Pull request merged**